src/transformState.js

```diff
@@ -6,6 +6,33 @@
  */
 function transformState(state, actions) {
   // write code here
+  for (const action of actions) {
+    const { type } = action;
+
+    switch (type) {
+      case 'addProperties':
+        const { extraData } = action;
+
+        Object.assign(state, extraData);
+        break;
+      case 'removeProperties':
+        const { keysToRemove } = action;
+
+        for (const key of keysToRemove) {
+          if (state.hasOwnProperty(key)) {
+            delete state[key];
+          }
+        }
+        break;
+      case 'clear':
+        for (const key in state) {
+          delete state[key];
+        }
+        break;
+      default:
+        break;
```

Comment:
For readability you could add spaces between cases 
```suggestion
      case 'addProperties':
        const { extraData } = action;

        Object.assign(state, extraData);
        break;
        
      case 'removeProperties':
        const { keysToRemove } = action;

        for (const key of keysToRemove) {
          if (state.hasOwnProperty(key)) {
            delete state[key];
          }
        }
        break;
        
      case 'clear':
        for (const key in state) {
          delete state[key];
        }
        break;
        
      default:
        break;
```